sdk/src/assertions/box_hash.rs

```diff
@@ -0,0 +1,431 @@
+// Copyright 2023 Adobe. All rights reserved.
+// This file is licensed to you under the Apache License,
+// Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+// or the MIT license (http://opensource.org/licenses/MIT),
+// at your option.
+
+// Unless required by applicable law or agreed to in writing,
+// this software is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR REPRESENTATIONS OF ANY KIND, either express or
+// implied. See the LICENSE-MIT and LICENSE-APACHE files for the
+// specific language governing permissions and limitations under
+// each license.
+
+use std::{fs::File, io::Cursor, path::*};
+
+use serde::{Deserialize, Serialize};
+use serde_bytes::ByteBuf;
+
+use crate::{
+    assertion::{Assertion, AssertionBase, AssertionCbor},
+    assertions::labels,
+    asset_io::{AssetBoxHash, CAIRead},
+    error::{Error, Result},
+    utils::hash_utils::{hash_stream_by_alg, verify_stream_by_alg, HashRange},
+    AssertionJson,
+};
+
+const ASSERTION_CREATION_VERSION: usize = 1;
+
+pub const C2PA_BOXHASH: &str = "C2PA";
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq)]
+pub struct BoxMap {
```

Comment:
I know this is just an implementation, it might be confusing to have these struct names opposite the schema defined in the specification (https://c2pa.org/specifications/specifications/1.3/specs/C2PA_Specification.html#_schema_and_example_5):

- Where `box-map` contains the `boxes`
- and `box-hash-map` contains box-names

Then on line 51 we have a `BoxHash`, which is playing games on my mind when comparing to the specification.

Reply:
BoxHash is the name of the Assertion.  This is what we call an assertion helper.  It builds a  c2p.hash.boxes.  This is where we make it easy to build and load the assertion for manipulation.   We do not expose raw structures.  This allows us to do things like accepts user JSON when what we are really doing is producing tagged CBOR.  We also do things like follow Rust naming conventions as opposed to those represented in the spec.